src/errors.ts

```diff
@@ -1,15 +1,15 @@
 import { TypedError } from 'typed-error';
 
 export class ObjectDoesNotExistError extends TypedError {}
+export class UniqueConstraintError extends TypedError {}
 export class UnauthorisedError extends TypedError {}
 export class BadRequestError extends TypedError {}
 export class ServerError extends TypedError {}
 
 export class HttpResponseError extends TypedError {
-	public statusCode: number;
-
-	constructor(message: string | Error, statusCode: number) {
+	constructor(message: string | Error, public statusCode: number) {
```

Comment:
I personally find this way of assigning instance attributes annoying because you no longer have attributes in one place -- some of them are declared in the class body, others indirectly in the constructor. This is less of a problem in this case given how small this class is, but it's still worth mentioning.